app/src/openxr/cpp/OpenXRHelpers.h

```diff
@@ -148,4 +150,25 @@ inline XrPosef MatrixToXrPose(const vrb::Matrix& aMatrix) {
     return result;
 }
 
+inline void ParseVersionString(const std::string& aString, int result[], int resultSize) {
+    std::istringstream stringStream(aString);
+    // Read the first number
+    stringStream >> result[0];
+    for(int i = 1; i < resultSize; i++) {
+        // Skip period
+        stringStream.get();
+        // Read the next number
+        stringStream >> result[i];
+    }
+}
+
+// This function compares two strings containing semantic versions in the MAJOR.MINOR.PATCH format.
+// It will return true if str1 is a lower version than str2, and false if it is the same or higher.
+inline bool CompareSemanticVersionStrings(const std::string& str1, const std::string& str2) {
+    int parsedStr1[3]{}, parsedStr2[3]{};
+    ParseVersionString(str1, parsedStr1, 3);
+    ParseVersionString(str2, parsedStr2, 3);
+    return std::lexicographical_compare(parsedStr1, parsedStr1 + 3, parsedStr2, parsedStr2 + 3);
+}
+
```

Comment:
I have created `SystemUtils.h` for this purpose, since we already have `DeviceUtils.h`, JNIUtil.h`, etc.

For now, `SystemUtils.h` contains the functions to handle the system version. We can move more functionality to that file in a different PR.